framework/include/reporters/ReporterContext.h

```diff
@@ -86,6 +108,51 @@ class ReporterContextBase : public libMesh::ParallelObject
                         const ReporterName & r_name,
                         unsigned int time_index = 0) const = 0;
 
+  /**
+   * Helper for enabling generic transfer of Reporter values to a vector
+   * @param r_data The ReporterData on the app that this data is being transferred to
+   * @param r_name The name of the Report being transfered to
+   *
+   * @see ReporterTransferInterface
+   */
+  virtual void transferToVector(ReporterData & r_data,
+                                const ReporterName & r_name,
+                                dof_id_type index,
+                                unsigned int time_index = 0) const = 0;
```

Comment:
Indeed, but like I said it might be difficult ensuring that the main-app value is a vector, unless it is explicitly declared with ReporterVectorContext, or possibly VectorPostprocessorContext.